### ISSUE TITLE
useChat not sending header options in request

### ISSUE THREAD

**Issue by apat183** [vercel/ai#2707]
### Description

I can't get the headers option to work in VueJS. It doesn't get passed (specifically "authorization". The "Content-Type" appears to be changing. I'm calling api on nodejs backend with express. but see screenshot below I don't think that matters as the request is not including the header. 

```
const { messages, input, handleSubmit } = useChat({
  api: '/api/ai/chat', headers: {
    'Content-Type': 'application/json',
    'Authorization': `Bearer ${getToken()}`
  }
});
```
I've tried creating new headers object:
```
const { messages, input, handleSubmit } = useChat({
  api: '/api/ai/chat', headers: new Headers({
    'Content-Type': 'application/json',
    'Authorization': getToken()
  })
});
```
As you can see in dev tools, nothing is added to request headers. On express side it's not received as well. 

<img width="351" alt="Screenshot 2024-08-17 at 10 44 02 AM" src="https://github.com/user-attachments/assets/5e408236-3460-4c29-b52f-79adb6f57d63">

### Code example
```
import { useChat } from '@ai-sdk/vue';
import { getToken } from '@/lib/supabase'

console.log(getToken())
const { messages, input, handleSubmit } = useChat({
  api: '/api/ai/chat', headers: new Headers({
    'Content-Type': 'application/json',
    'Authorization': getToken()
  })
});
```

### Additional context

Just add, this works with own code and using fetch. So can't see issue on backend or through other code as I'm just changing the request/tool 

```
const response = await fetch('/api/ai/chat', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
      Authorization: getToken()
    },
    body: JSON.stringify({
      prompt: prompt
    })
  })
```

**Comment by apat183** [comment]
Update in some investigation.

Looks like useCompletion works. 

I noticed in the use-chat.ts. The headers and body are never used? 

<img width="561" alt="Screenshot 2024-08-17 at 11 54 09 AM" src="https://github.com/user-attachments/assets/dbd014ea-a749-4207-bf7d-2a828817a470">

**Comment by lgrammel** [comment]
Fix: https://github.com/vercel/ai/pull/2718

**Comment by lgrammel** [comment]
Fixed in `@ai-sdk/vue@0.0.38`

**Comment by lgrammel** [comment]
## Identify issue type

Identified as **Bug**  
Confidence: **High**

### Rationale
This is a confirmed defect in existing Vue `useChat` behavior: supported request options were accepted but not applied, and the maintainers released a fix.

### Evidence
- Vue `useChat` accepted a `headers` option but omitted the configured `Authorization` header from its request, while a direct `fetch` and `useCompletion` worked.
- The reporter found that `headers` and `body` were not used in the Vue `useChat` implementation.
- A maintainer linked PR #2718 as the fix and confirmed it was released in `@ai-sdk/vue@0.0.38`.
